### ISSUE TITLE
Adding footnotes

### ISSUE THREAD

**Issue by bakerolls** [vito/booklit#16]
Hi, I like to use footnotes on my pages. I wrote [a very simple plugin](https://github.com/bakerolls/booklit-footnotes) that accomplishes this by adding arguments to a call of `\footnote` to a package global slice and rendering it when `\footnotemark` is called. This is probably not the best way to do it.

Do you think a feature like this should be added to `baselit`?

**Comment by vito** [comment]
Neat! Yeah, this could be a good idea for `baselit`.

Maybe the footnotes could be kept as state on the `*Plugin` object instead?

**Comment by bakerolls** [comment]
To be honest, I don't know why I didn't set it inside the struct. The repo is updated, thank you. There still is the problem of duplicate references when using multiple `\footnotemark` on a single page (which might not happen often).

For example, both footnotes will get an index `0`:

```lit
\footnote{Foo.}
\footnotemark

\footnote{Bar.}
\footnotemark
```

**Comment by vito** [comment]
(Sorry for the delay!)

For my understanding, do you see this to mainly be a problem when multiple *sections* are rendered inline into one page?

Maybe the footnotes should include the section tag in their anchor - i.e. `#fn-section-title-1`?

**Comment by bakerolls** [comment]
Exactly, I think your solution should work. How could this be implemented? As far as I know, you can't access the current sections title from within `booklit.Styled`.

**Comment by bakerolls** [comment]
Sorry, this took longer than expected.

I am not sure how to access previous footnotes from within `baselit`. The plugin-approach seems a lot simpler.

I suspect it would work similar to the toc, but reading upward and printing the sections titles querying the footnotes in all subsections.

A new `\footnote` then should look in its parent sections until it finds a previous `\footnote` and increment the found id or it finds a `\footnotemark`, in which case its id should start at 0 or 1.

**Comment by vito** [comment]
@bakerolls No worries! I think a separate plugin is fine especially since there's state involved.

I think if you put the section's tag in the footnote anchor the IDs don't really matter, since they'll always be relative to the section tag. Does that make sense?
